src/DocumentationViewer.js

```diff
@@ -110,117 +108,104 @@ export default function DocumentationViewer({ application }) {
     open();
   });
   application.add_action(action_documentation);
 }
 
-function createSearchListModel(base_path, indexes, webview, filter) {
-  const model = newListStore();
-  for (let i = 0; i < indexes.length; i++) {
-    const index = indexes[i];
-    const dir = base_path.get_child(index.dir);
-    const meta = index.meta;
-    const namespace = `${meta.ns}-${meta.version}`;
-    model.append(
-      new DocumentationPage({
-        uri: dir.get_child("index.html").get_uri(),
-        search_name: namespace,
-      }),
-    );
-    for (const symbol of index.symbols) {
-      model.append(
-        new DocumentationPage({
-          search_name: getSearchNameForDocument(symbol, meta),
-          uri: `${dir.get_uri()}/${getLinkForDocument(symbol)}`,
-        }),
-      );
-    }
-  }
-  return createSearchSelectionModel(model, filter, webview);
-}
-
-function createSearchSelectionModel(model, filter, webview) {
-  const filter_model = Gtk.FilterListModel.new(model, filter);
-  const sorter = Gtk.StringSorter.new(filter.expression);
-  const sort_model = Gtk.SortListModel.new(filter_model, sorter);
-  const selection_model = Gtk.SingleSelection.new(sort_model);
+async function loadLibrary(model, directory) {
+  try {
+    const json_file = directory.get_child("index.json");
+    const html_file = directory.get_child("index.html");
 
-  selection_model.connect("selection-changed", () => {
-    const uri = selection_model.selected_item.uri;
-    webview.load_uri(uri);
-  });
-  return selection_model;
-}
+    const [data] = await json_file.load_contents_async(null);
+    const index = JSON.parse(decode(data));
 
-function createBrowseListModel(base_path, indexes, webview) {
-  const model = newListStore();
-  for (let i = 0; i < indexes.length; i++) {
-    const index = indexes[i];
-    const dir = base_path.get_child(index.dir);
     const namespace = `${index.meta.ns}-${index.meta.version}`;
-    model.append(
-      new DocumentationPage({
-        name: namespace,
-        uri: dir.get_child("index.html").get_uri(),
-        children: getChildren(index, dir),
-      }),
-    );
+    const page = new DocumentationPage({
+      name: namespace,
+      search_name: namespace,
+      uri: html_file.get_uri(),
+      children: getChildren(index, directory),
+    });
+
+    model.append(page);
+  } catch (error) {
+    if (!error.matches(Gio.IOErrorEnum, Gio.IOErrorEnum.NOT_FOUND))
+      logError(error);
   }
-  return createBrowseSelectionModel(model, webview);
 }
 
-async function createIndex(base_path, dirs) {
-  const indexes = [];
-  for (const dir of dirs) {
-    indexes.push(readIndexJSON(base_path, dir));
+const IGNORED_LIBRARIES = [
+  "atk",
+  "javascriptcoregtk-4.1",
+  "libhandy-1",
+  "libnotify-0",
+  "webkit2gtk-4.1",
+  "webkit2gtk-web-extension-4.1",
+];
+
+async function scanLibraries(model, base_dir) {
+  const libraries = [];
+
+  const iter = await base_dir.enumerate_children_async(
+    "standard::*",
```

Comment:
```suggestion
    "standard::name,standard::type",
```

Since we're here to optimize.